src/pages/patientView/ctDNA/HeatmapVAFEvents.tsx

```diff
@@ -96,17 +37,62 @@ export default class HeatmapVAFEvents extends React.Component<IHeatmapVAFEventsP
       let mutSample = mutation.sampleId;
       if (~sampleOrder.indexOf(mutSample) && ~heatmapGeneList.indexOf(mutGene)) {
         let vaf = mutation.tumorAltCount/(mutation.tumorRefCount + mutation.tumorAltCount);
-        let mutPush = {
-          x: sampleOrder.indexOf(mutSample),
-          y: heatmapGeneList.indexOf(mutGene),
-          z: Math.round(vaf * 1000),
-        };
+        let mutPush = [
+          sampleOrder.indexOf(mutSample),
+          heatmapGeneList.indexOf(mutGene),
+          vaf,
+        ];
         plotData.push(mutPush);
       }
     });
 
+    let plotMaker = {
+      data: plotData,
+      xTicks: sampleOrder,
+      yTicks: heatmapGeneList,
+    };
+    return plotMaker;
+  }
+
+  componentDidMount() {
+    let plotMaker = this.extractPlotData();
+
+    let config = {};
+
+    let data = {
+      labels : ['0h','1h','2h','3h','4h','5h','6h','7h','8h','9h','10h','11h'],
+      datasets : [
+        {
+          label: 'Monday',
+          data: [8, 6, 5, 7, 9, 8, 1, 6, 3, 3, 8, 7]
+        },
+        {
+          label: 'Tuesday',
+          data: [6, 8, 5, 6, 5, 5, 7, 0, 0, 3, 0, 7]
+        },
+        {
+          label: 'Wednesday',
+          data: [8, 5, 6, 4, 2, 2, 3, 0, 2, 0, 10, 8]
+        },
+        {
+          label: 'Thursday',
+          data: [4, 0, 7, 4, 6, 3, 2, 4, 2, 10, 8, 2]
+        },
+        {
+          label: 'Friday',
+          data: [1, 0, 0, 7, 0, 4, 1, 3, 4, 5, 1, 10]
+        }
+      ]
+    };
+
+    var canvas:any = document.getElementById("heatmap");
```

Comment:
we should never use id's in this manner because then there can only be one instance of this on a page.  it is not self-contained as it relies on document global.   instead use React refs to get a reference to <canvas> element. https://facebook.github.io/react/docs/refs-and-the-dom.html